### ISSUE TITLE
Port for vcpkg

### ISSUE THREAD

**Issue by Nosenzor** [elalish/manifold#853]
Hello,
I've opened an issue in the [VCPKG repo[]( https://github.com/microsoft/vcpkg/issues/38066) to get this nice library integrated in the packaging system that i use most often. It would be very convenient to have Manifold supported by vcpkg. 
If any one feel like he can do it that would be super nice.
If noone here can do it soon, I will try to do it myself but that would be in a month or so.
PS : i'm also using manifold through the python binding, i may later open some issues we have met so far.
Regards !

**Comment by elalish** [comment]
Thank you! Yes, availability in more packaging systems would be great, especially if it can be automated in our CI. We're currently working toward a v3.0 release where we are planning to significantly reduce our dependencies (e.g. removing Thrust), so it might be easier to package once that's ready. We would certainly appreciate the help since none of us are Windows users and don't have much familiarity with vcpkg.

**Comment by pca006132** [comment]
It should be much simpler to port this to vcpkg now (after #857 and maybe #810). I think we only need tbb and glm, clipper2 and quickhull are optional.

**Comment by petamas** [comment]
> We would certainly appreciate the help since none of us are Windows users and don't have much familiarity with vcpkg.

Just wanted to mention that vcpkg is multiplatform, not tied to Windows. It is available on Windows, Mac, Linux, QNX, etc.

**Comment by petamas** [comment]
> I think we only need tbb and glm, clipper2 and quickhull are optional.

I think you might be mistaken, clipper2 and glm are mandatory dependencies (if `find_package()` cannot find them, `FetchContent` is used to add them), and so is quickhull (the submodule is `add_subdirectory()`'d unconditionally). TBB and nanobind are optional (the project does not need them with the default options).

**Comment by elalish** [comment]
quickhull is on the verge of being removed entirely - that might even land this week. clipper2 is optional as you can choose to build `Manifold` and not `CrossSection`, but I'd guess we'd rather have both for vcpkg. And yes, it's great that vcpkg is cross-platform, though I think it may have a bit more adoption on Windows...

**Comment by elalish** [comment]
True, I don't think we've added that as an option in our CMake script itself yet - probably not a bad idea. Any interest?

**Comment by Nosenzor** [comment]
Hey guys, how is this going on? It seems you did huge progress’ do you think you are ready for vcpkg support?

**Comment by pca006132** [comment]
Hi, now that v3.0 is released, this should be easy to port to vcpkg.

**Comment by JeffreyWardman** [comment]
Has anyone picked this up yet? Otherwise, happy to have a crack at it.

**Comment by pca006132** [comment]
@JeffreyWardman I don't think anyway picked this up. Please let us know if you encountered any problem, and feel free to open a PR to add a link in our readme later :)

**Comment by JeffreyWardman** [comment]
I've added (and tested on windows only) the port without issues! Active PR: https://github.com/microsoft/vcpkg/pull/42347

**Comment by BasTossings** [comment]
I also looked at @weianweigan his NuGet GitHub action and I can't see what he is doing differently to get it to build. It's also building an older version of Manifold so it could be that.

**Comment by JeffreyWardman** [comment]
Hi all, it's been merged into vcpkg now! As of: https://github.com/microsoft/vcpkg/commit/c7e10f6304b98e9b2f7d6d40446053398da0698f

I added a very basic mention in the readme: https://github.com/elalish/manifold/pull/1108
